common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -667,6 +667,17 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         + " table or partition level. If hive.exec.parallel \n"
         + "is set to true then max worker threads created for copy can be hive.exec.parallel.thread.number(determines \n"
         + "number of copy tasks in parallel) * hive.repl.parallel.copy.tasks "),
+    REPL_SNAPSHOT_DIFF_FOR_EXTERNAL_TABLE_COPY("hive.repl.externaltable.snapshotdiff.copy",
+        false,"Use snapshot diff for copying data from source to "
+        + "destination cluster for external table in distcp"),
+    REPL_SNAPSHOT_PREFIX_FOR_EXTERNAL_TABLE_COPY("hive.repl.externaltable.snapshot.prefix",
```

Comment:
can we have some default value here? Or if its mandatory value, we should fail at policy creation itself

Reply:
Added validation in the code, Extended a test as well to confirm the validation. Making it user defined to prevent collision with any other distCp job with similar names.